libecole/tests/src/dynamics/test-primalsearch.cpp

```diff
@@ -0,0 +1,209 @@
+}
+
+TEST_CASE("PrimalSearchDynamics functional tests", "[dynamics]") {
+	const auto trials_per_node = 5;
+	auto dyn = dynamics::PrimalSearchDynamics{trials_per_node};
+	auto model = get_model();
+
+	SECTION("Return valid action set") {
+		auto const [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		REQUIRE(var_ids.size() <= model.variables().size());
+		REQUIRE(xt::all(var_ids >= 0));
+		REQUIRE(xt::all(var_ids < model.variables().size()));
+		REQUIRE(xt::unique(var_ids).size() == var_ids.size());
+	}
+
+	SECTION("Handle extreme values - empty") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		std::tie(done, action_set) = dyn.step_dynamics(model, {});
+		REQUIRE_FALSE(done);
+	}
+
+	SECTION("Handle extreme values - min") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		auto const& action = filled_action(action_set, std::numeric_limits<SCIP_Real>::min());
+		std::tie(done, action_set) = dyn.step_dynamics(model, action);
+		REQUIRE_FALSE(done);
+	}
+
+	SECTION("Handle SCIP special values - SCIP_UNKNOWN") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		auto const& action = filled_action(action_set, SCIP_UNKNOWN);
+		std::tie(done, action_set) = dyn.step_dynamics(model, action);
+		REQUIRE_FALSE(done);
+	}
+
+	SECTION("Handle SCIP special values - SCIP_INVALID") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		auto const& action = filled_action(action_set, SCIP_INVALID);
+		std::tie(done, action_set) = dyn.step_dynamics(model, action);
+		REQUIRE_FALSE(done);
+	}
+
+	SECTION("Handle SCIP special values - infinity") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		auto const& action = filled_action(action_set, SCIPinfinity(model.get_scip_ptr()));
+		std::tie(done, action_set) = dyn.step_dynamics(model, action);
+		REQUIRE_FALSE(done);
+	}
+
+	SECTION("Handle SCIP special values - minus infinity") {
+		auto [done, action_set] = dyn.reset_dynamics(model);
+		REQUIRE(action_set.has_value());
+		auto const& var_ids = action_set.value();
+		REQUIRE(var_ids.size() > 0);
+		auto const& action = filled_action(action_set, -SCIPinfinity(model.get_scip_ptr()));
+		std::tie(done, action_set) = dyn.step_dynamics(model, action);
+		REQUIRE_FALSE(done);
+	}
```

Comment:
I think you could regroup all these sections in one, 
```suggestion
	SECTION("Handle extreme values") {
		auto [done, action_set] = dyn.reset_dynamics(model);
		REQUIRE(action_set.has_value());
		auto const& var_ids = action_set.value();
		REQUIRE(var_ids.size() > 0);
		auto const inf = SCIPinfinity(model.get_scip_ptr());
		auto const val = GENERATE(std::numeric_limits<SCIP_Real>::min(), SCIP_UNKNOWN, SCIP_INVALID, -inf, inf);
		auto const& action = filled_action(action_set, val);
		std::tie(done, action_set) = dyn.step_dynamics(model, action);
		REQUIRE_FALSE(done);
	}
```

Reply:
Cleaner indeed. Thanks !